### PR TITLE
feat: add an option to skip downed peers in KubeSpan

### LINKED ISSUE #4248: configurable policy for kubespan enforcement for down/unknown peers
## Feature Request

The question is: should we force traffic to go via KubeSpan even if the peer is not up? 

If we do, it means traffic is always encrypted (as soon as discovery catches up though), if we don't traffic might silently become not encrypted and go directly to the peer if the wireguard connection is down.

This should be configurable with default to force encryption.

### Description

Implementation-wise: should we add allowedIPs of the downed/unknown peers to the nftables.

### PR DESCRIPTION
**Pull request by smira**
Fixes #4248

This resolves the balance between security and connectivity.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4301)

### CODE REVIEW

File: website/content/docs/v0.13/Reference/configuration.md [hunk 1]
```diff
@@ -5262,6 +5262,21 @@ Cluster discovery should be enabled with .cluster.discovery.enabled for KubeSpan
 </div>
 
 <hr />
+<div class="dd">
+
+<code>skipDownPeers</code>  <i>bool</i>
```
Comment (smira) [inline website/content/docs/v0.13/Reference/configuration.md:5267]: @Ulexus any idea for a better parameter name?

**Review by rsmitty — APPROVED** [review]

File: website/content/docs/v0.13/Reference/configuration.md (hunk 1 above)
Comment (smira) [reply website/content/docs/v0.13/Reference/configuration.md:5267]: from @Ulexus: `AllowDownPeerBypass`

File: pkg/machinery/config/provider.go [hunk 2]
```diff
@@ -217,6 +217,7 @@ type Route interface {
 // KubeSpan configures KubeSpan feature.
 type KubeSpan interface {
 	Enabled() bool
+	ForceRouting() bool
```
Comment (Ulexus) [inline pkg/machinery/config/provider.go:220]: Should this naming now mirror the config flag?

**Review by Ulexus — COMMENTED** [review]
Align config flag, CLI flag, and config machinery accessor function name?

**Pull request merged**